### ISSUE TITLE
Add support for Ed25519-SHA256

### ISSUE THREAD

**Issue by travisspencer** [markenwerk/java-utils-mail-dkim#26]
Right now, it is only possible to sign emails using RSA keys with this library. From the little I know about DKIM, it's also possible to use Elliptic curve keys, as defined in RFC 8463. When given the choice, a lot of people would opt for EC over RSA (smaller keys, more secure depending on input, etc.).

To this end, it would be nice if this library also supported Elliptic curve keys in a way that complies with this RFC.

**Comment by toKrause** [comment]
Hi, 

sorry for the delayed reaction. I don't really have much time to maintain this library. 

You brought up an interesting issue. I wasn't aware of RFC 8463 until now and as much as I can tell, a lot of DKIM related software and available tools haven't added support so far. 

I tried to find out how well Ed25519 is supported by Java. According to [this release note](https://openjdk.java.net/projects/jdk/15/) from OpenJDK at least, i gather that i will be available by default as of Java 15. I think it would be optimistic to assume that Java 15 is a common execution environment for current Java software. I also found [this library](https://github.com/str4d/ed25519-java).  there hasn't been much activity lately, but it seems to work fine. 

I've implemented support for Ed25519-SHA256 using that library, **but I'd call that an experimental feature for now**. The library comes with an JCE provider for Ed25519 that integrates into the JCE API, I wasn't quite able to get rid of all direct usages of that library. Maybe my knowledge about JCE isn't quite extensive enough.

Also, I suspect that the original author of this DKIM library didn't really expect there to be different types of key pair to be used. RSA key pairs were quite hard coded into the library.

Up until now `DkimSigner` had a primary constructor that took a `RSAPrivateKey` and two convenience constructors that 
took an `InputStream` or a `File`. The convenience constructors load an RSA private key that is PKCS#8 encoded and DER formatted. 

I don't know quite enough about the typically used storage formats for Ed25519 private and public keys or the tools typically used to create them, so I can't provide similar convenience constructors for Ed25519 private keys.

Using the Ed25519 key pair from the example section of RFC 8463 and the above mentioned Ed25519 library, it is now possible to create a `DkimSigner`  with an `EdDSAPrivateKey` instead of an `RSAPrivateKey`. But the `EdDSAPrivateKey` has to be created manually.

``` java
// add the jce provider from the ed25519 library
Security.addProvider(new EdDSASecurityProvider());

// create a ed25519 private key specification using the example private key
KeyFactory keyFactory = KeyFactory.getInstance("EdDSA");
byte[] privKeyBytes = Base64.decode("nWGxne/9WmC6hEr0kuwsxERJxWl7MmkZcDusAxyuf2A=");
EdDSAPrivateKeySpec privateKeySpec = new EdDSAPrivateKeySpec(privKeyBytes, EdDSANamedCurveTable.ED_25519_CURVE_SPEC);

// create a ed25519 private key from the private key specification
EdDSAPrivateKey privateKey = (EdDSAPrivateKey) keyFactory.generatePrivate(privateKeySpec);

// create a skim signer for a domain, a selector ant the private key
DkimSigner dkimSigner = new DkimSigner("example.com", "selector", privateKey);
```
As mentioned above, I don't know enough about  typically used storage formats for Ed25519 private and public keys or the tools typically used to create them. Therefore I've only tested the new features with the example key pair from RFC 8463. I've also tried creating another key pair using `ssh-keygen -t ed25519`, but key pairs that are created with this tool seem to be stored in a different format that isn't directly compatible with the Ed25519 library.

I will release an updated version of the library to Maven Central shortly.

I hope these changes will be of any use an I would appreciate any feedback. But I can't promise that i will find time to do further updates on short notice. 

Best regards.
